### PR TITLE
[AccountAddress][1/3] use the latest move

### PR DESCRIPTION
**Pull request by areshand**
### [This is a duplicate of the previous PR to private repo https://github.com/aptos-labs/aptos-core/pull/135]

use latest move to unblock the further address20 or address32 migration. 
The change includes

**use the latest revision of move in toml**
**address some breaking changes introduced by the following commits**
* [PR update the intital gas schedule to initial cost schedule](https://github.com/diem/move/commit/1a039e4e3ffb91d902bf2eaefa77e52177b8ec6f)
* [PR update the interface for move Complier](https://github.com/diem/move/commit/bc6c645db948ddb6f9acec3174826d2f71e58447)
* a regex version update to 1.5.5 
** add a workaround to the language_storage types

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
Cargo test passed except  except one flaky test

```
failures:
    persistent_safety_storage::tests::test_waypoint_counters

test result: FAILED. 12 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 11.03s
```

### CODE REVIEW

**Comment by davidiw** [comment]
nice, this actually seems like it is good, but I noticed that this test "framework::generated_files test_that_generated_file_are_up_to_date_in_git" didn't quite make it. can you double check locally and see if you're missing some files you should have compiled?

my guess is aptos-move/transaction-builder-generator/

**Review by davidiw — APPROVED** [review]

File: testsuite/generate-format/tests/staged/consensus.yaml [hunk 1]
```diff
@@ -437,7 +437,7 @@ TwoChainTimeoutCertificate:
 TypeTag:
   ENUM:
     0:
-      Bool: UNIT
+      bool: UNIT
```
[the remark this answers is not in the document]
Comment (areshand) [reply testsuite/generate-format/tests/staged/consensus.yaml:440]: I noticed this change as well.  I didn't figure out why this become lowercase. But thought this lower case is not breaking anything and didn't debug further to find cause.

**Pull request closed without merging**